File: BookStore_API/BookStore_API/Models/Tinnhanhotro.cs
using System;
using System.Collections.Generic;

namespace BookStore_API.Models;

public partial class Tinnhanhotro
{
    public int Matinnhan { get; set; }

    public int Mahoidap { get; set; }

    public string Nguoigui { get; set; } = null!;

    public int? Makh { get; set; }

    public int? Manv { get; set; }

    public string Noidung { get; set; } = null!;

    public string? Hinhanh { get; set; }

    public bool? Daxem { get; set; }

    public DateTime? Thoigian { get; set; }

    public virtual Hoidap MahoidapNavigation { get; set; } = null!;

    public virtual Khachhang? MakhNavigation { get; set; }

    public virtual Nhanvien? ManvNavigation { get; set; }
}

File: thuvienapp/lib/models/tin_nhan_ho_tro.dart
class TinNhanHoTro {
  final int maTinNhan;
  final int maHoiDap;
  final String nguoiGui; // "KhachHang" or "NhanVien"
  final int? maKh;
  final int? maNv;
  final String noiDung;
  final String? hinhAnh;
  final bool daXem;
  final DateTime thoiGian;

  TinNhanHoTro({
    required this.maTinNhan,
    required this.maHoiDap,
    required this.nguoiGui,
    this.maKh,
    this.maNv,
    required this.noiDung,
    this.hinhAnh,
    required this.daXem,
    required this.thoiGian,
  });

  factory TinNhanHoTro.fromJson(Map<String, dynamic> json) {
    return TinNhanHoTro(
      maTinNhan: json['matinnhan'] ?? 0,
      maHoiDap: json['mahoidap'] ?? 0,
      nguoiGui: json['nguoigui'] ?? 'KhachHang',
      maKh: json['makh'],
      maNv: json['manv'],
      noiDung: json['noidung'] ?? '',
      hinhAnh: json['hinhanh'],
      daXem: json['daxem'] ?? false,
      thoiGian: json['thoigian'] != null
          ? DateTime.parse(json['thoigian'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'matinnhan': maTinNhan,
      'mahoidap': maHoiDap,
      'nguoigui': nguoiGui,
      'makh': maKh,
      'manv': maNv,
      'noidung': noiDung,
      'hinhanh': hinhAnh,
      'daxem': daXem,
      'thoigian': thoiGian.toIso8601String(),
    };
  }
}
